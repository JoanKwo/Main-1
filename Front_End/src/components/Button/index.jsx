import React from "react";
import styled from "styled-components";

const RedButtonStyle = styled.button`
    color: #fff !important;
    text-transform: uppercase;
    text-decoration: none;
    background: #cc2344;
    padding: 10px;
    border-radius: 10px;
    display: inline-block;
    border: none;
    margin: 10px;
    box-sizing: border-box
    transition: all 0.4s ease 0s;
    :hover {
        background: #434343;
        letter-spacing: 1px;
        -webkit-box-shadow: 0px 5px 40px -10px rgba(0, 0, 0, 0.57);
        -moz-box-shadow: 0px 5px 40px -10px rgba(0, 0, 0, 0.57);
        box-shadow: 5px 40px -10px rgba(0, 0, 0, 0.57);
        transition: all 0.4s ease 0s;
      }
    `;
const BlueButtonStyle = styled.button`
    color: #Fff !important;
    text-transform: uppercase;
    text-decoration: none;
    background: #0000FF;
    padding: 10px;
    border-radius: 10px;
    display: inline-block;
    border: none;
    margin: 10px;
    box-sizing: border-box
    transition: all 0.4s ease 0s;
    :hover {
        background: #434343;
        letter-spacing: 1px;
        -webkit-box-shadow: 0px 5px 40px -10px rgba(0, 0, 0, 0.57);
        -moz-box-shadow: 0px 5px 40px -10px rgba(0, 0, 0, 0.57);
        box-shadow: 5px 40px -10px rgba(0, 0, 0, 0.57);
        transition: all 0.4s ease 0s;
      }
    `;
export default function Button({ onClickAction, buttonText, buttonColor }) {
  if (buttonColor === "red") {
    return (
      <RedButtonStyle onClick={onClickAction}>{buttonText}</RedButtonStyle>
    );
  } else if (buttonColor === "blue") {
    return (
      <BlueButtonStyle onClick={onClickAction}>{buttonText}</BlueButtonStyle>
    );
  }
  return <button onClick={onClickAction}>{buttonText}</button>;
}
