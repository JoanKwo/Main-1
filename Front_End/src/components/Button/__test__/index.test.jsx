import React from "react";
import Button from "../index";
import renderer from "react-test-renderer";
import "jest-styled-components";

describe("Button tests", () => {
  it("button works", () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("testing with color", () => {
    const redButton = renderer.create(<Button buttonColor={"red"} />).toJSON();
    expect(redButton).toMatchSnapshot();
    const blueButton = renderer.create(<Button buttonColor={"red"} />).toJSON();
    expect(blueButton).toMatchSnapshot();
    const noColorButton = renderer.create(<Button />).toJSON();
    expect(noColorButton).toMatchSnapshot();
  });
});
