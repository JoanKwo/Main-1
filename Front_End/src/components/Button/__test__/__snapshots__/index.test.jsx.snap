// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests button works 1`] = `<button />`;

exports[`Button tests testing with color 1`] = `
.c0 {
  color: #fff !important;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #cc2344;
  padding: 10px;
  border-radius: 10px;
  display: inline-block;
  border: none;
  margin: 10px;
  box-sizing: border-box transition:all 0.4s ease 0s;
}

.c0:hover {
  background: #434343;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
  -moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
  box-shadow: 5px 40px -10px rgba(0,0,0,0.57);
  -webkit-transition: all 0.4s ease 0s;
  transition: all 0.4s ease 0s;
}

<button
  className="c0"
/>
`;

exports[`Button tests testing with color 2`] = `
.c0 {
  color: #fff !important;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #cc2344;
  padding: 10px;
  border-radius: 10px;
  display: inline-block;
  border: none;
  margin: 10px;
  box-sizing: border-box transition:all 0.4s ease 0s;
}

.c0:hover {
  background: #434343;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
  -moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
  box-shadow: 5px 40px -10px rgba(0,0,0,0.57);
  -webkit-transition: all 0.4s ease 0s;
  transition: all 0.4s ease 0s;
}

<button
  className="c0"
/>
`;

exports[`Button tests testing with color 3`] = `<button />`;
